src/OpenTelemetry.Api/Logs/LoggerProvider.cs

```diff
@@ -25,33 +29,53 @@ internal class LoggerProvider : BaseProvider
 {
     private static readonly NoopLogger NoopLogger = new();
 
-    /// <summary>
-    /// Initializes a new instance of the <see cref="LoggerProvider"/> class.
-    /// </summary>
-    protected LoggerProvider()
-    {
-    }
-
     /// <summary>
     /// Gets a logger.
     /// </summary>
     /// <returns><see cref="Logger"/> instance.</returns>
     public Logger GetLogger()
-        => this.GetLogger(new InstrumentationScope());
+        => this.GetLogger(name: null, version: null);
 
     /// <summary>
     /// Gets a logger with the given name.
     /// </summary>
     /// <param name="name">Optional name identifying the instrumentation library.</param>
     /// <returns><see cref="Logger"/> instance.</returns>
-    public Logger GetLogger(string name)
-        => this.GetLogger(new InstrumentationScope(name));
+    public Logger GetLogger(string? name)
+        => this.GetLogger(name, version: null);
+
+    /// <summary>
+    /// Gets a logger with the given name and version.
+    /// </summary>
+    /// <param name="name">Optional name identifying the instrumentation library.</param>
+    /// <param name="version">Optional version of the instrumentation library.</param>
+    /// <returns><see cref="Logger"/> instance.</returns>
+    public Logger GetLogger(string? name, string? version)
+    {
+        if (!this.TryCreateLogger(name, out var logger))
+        {
+            return NoopLogger;
+        }
+
+        logger!.SetInstrumentationScope(version);
+
+        return logger;
+    }
 
     /// <summary>
-    /// Gets a logger with given instrumentation scope.
+    /// Try to create a logger with the given name.
     /// </summary>
-    /// <param name="instrumentationScope"><see cref="InstrumentationScope"/>.</param>
-    /// <returns><see cref="Logger"/>.</returns>
-    public virtual Logger GetLogger(InstrumentationScope instrumentationScope)
-        => NoopLogger;
+    /// <param name="name">Optional name identifying the instrumentation library.</param>
+    /// <param name="logger"><see cref="Logger"/>.</param>
+    /// <returns><see langword="true"/> if the logger was created.</returns>
+    protected virtual bool TryCreateLogger(
+        string? name,
+#if NETSTANDARD2_1_OR_GREATER || NET6_0_OR_GREATER
```

Comment:
A note about this condition...

`OpenTelemetry.Api` currently targets `netstandard2.0;net462` so this will never resolve to true for any binary in the package. @TimothyMothra is currently exploring how to best approach this (eg #4293). Once we have a more recent target (or two) this should light up. In the meantime nothing catastrophic will happen just some wonky false-positive warnings about possibly accessing a null variable.